src/main/java/blang/distributions/Generators.java

```diff
@@ -25,6 +28,64 @@
 /** Various random number generators. */
 public class Generators // Warning: blang.distributions.Generators hard-coded in ca.ubc.stat.blang.scoping.BlangImplicitlyImportedFeatures 
 {
+  /** */
+  public static double gompertz(Random rand, double shape, double scale)
+  {
+    double percentile = rand.nextDouble();
+    //double result = - (Math.log(shape) - Math.log(shape - Math.log(1 - percentile)*Math.log(scale))) / Math.log(scale);
+    double result = scale * Math.log(1 - (1 / shape)*Math.log(1-percentile));
+    if (result == 0.0)
+    		result = ZERO_PLUS_EPS;
+    return result;
+  }
+  
+  /** */
+  public static double gumbel(Random rand, double location, double scale)
+  {
+    double result = new GumbelDistribution(generator(rand), location, scale).sample();
+    if (result == 0.0) 
+    		result = ZERO_PLUS_EPS;
+    return result;
+  }
+  
+  /** */
+  public static double weibull(Random rand, double scale, double shape)
+  {
+    double percentile = rand.nextDouble();
+    double result = scale*Math.pow(-Math.log(1 - percentile), (1/shape));
+    if (result == 0.0)
+    		result = ZERO_PLUS_EPS;
+    return result;
+  }
+  
+  /** */
+  public static double fDist(Random rand, double d1, double d2) 
+  {
+	double result = new FDistribution(generator(rand), d1, d2).sample();
+	if (result == 0.0)
+		result = ZERO_PLUS_EPS;
+	return result;
+  }
+  
+  /** */
+  public static double logisticDist(Random rand, double location, double scale) 
+  {
+	double result = new LogisticDistribution(generator(rand), location, scale).sample();
+	if (result == 0.0)
```

Comment:
The logistic distribution has positive density at zero, so please remove this if here